test/gtest/uct/ib/test_ud_ds.cc

```diff
@@ -52,21 +52,41 @@ class test_ud_ds : public uct_test {
     }
 
     void test_cep_insert(entity *e, uct_ib_address_t *ib_addr,
-                         uct_ud_iface_addr_t *if_addr, unsigned base);
+                         uct_ud_iface_addr_t *if_addr, unsigned base,
+                         unsigned num_paths);
 
     void create_and_insert_ep(entity *e, uct_ib_address_t *ib_addr,
                               uct_ud_iface_addr_t *if_addr,
-                              uct_ud_ep_conn_sn_t conn_sn, unsigned ep_index,
-                              bool ep_private);
+                              unsigned path_index, uct_ud_ep_conn_sn_t conn_sn,
+                              unsigned ep_index, bool ep_private);
 
     uct_ud_ep_t *get_ep(entity *e, uct_ib_address_t *ib_addr,
-                        uct_ud_iface_addr_t *if_addr,
+                        uct_ud_iface_addr_t *if_addr, unsigned path_index,
                         uct_ud_ep_conn_sn_t conn_sn, bool ep_private);
 
     void insert_ep(entity *e, uct_ib_address_t *ib_addr,
-                   uct_ud_iface_addr_t *if_addr,
+                   uct_ud_iface_addr_t *if_addr, unsigned path_index,
                    uct_ud_ep_conn_sn_t conn_sn, uct_ud_ep_t *ep);
 
+    ucs_conn_sn_t get_conn_sn(uct_ud_iface_t *ud_iface, size_t i,
+                              unsigned num_paths, unsigned path_index)
+    {
+        if (uct_ib_iface_is_roce(&ud_iface->super)) {
+            if (ud_iface->super.config.roce_path_factor == 0) {
```

Comment:
why needed? let's test only roce_path_factor != 0

Reply:
do you mean to skip the case if `roce_path_factor == 0`? or not set it in the test?

Reply:
just have no special check for it in the test

Reply:
ok, removed